### ISSUE TITLE
 No unread! method for HTTP.ConnectionPool.Transaction{MbedTLS.SSLContext}!

### ISSUE THREAD

**Issue by pevnak** [JuliaWeb/HTTP.jl#340]
Hi,

I am getting this warning.
```
WARNING: No unread! method for HTTP.ConnectionPool.Transaction{MbedTLS.SSLContext}!
         Discarding 9233 bytes!
```
is this OK, or it is something I should be afraid of?

Then, I sometimes occasionally get this error.
```IOError(EOFError() during request(https://cta-features-jld.s3.eu-west-1.amazonaws.com/url20171004/0-part-00003.gz))```
Can I prevent this?

Thank you very much for an answer.

Tomas

P.S. I use Julia 0.7 with this list of packages.

Project ReadData v0.1.0
    Status `Project.toml`
  [4f1ea46c] AWSCore v0.5.1+ #master (https://github.com/JuliaCloud/AWSCore.jl.git)
  [1c724243] AWSS3 v0.4.0
  [944b1d66] CodecZlib v0.5.0
  [a93c6f00] DataFrames v0.14.1
  [5789e2e9] FileIO v1.0.2
  [53afe959] FlatBuffers v0.4.0
  [132d396a] HTTP v0.7.0 #so/lazyintegrate (https://github.com/JuliaWeb/HTTP.jl.git)
  [50d2b5c4] Lazy v0.13.1
  [9920b226] MLDataPattern v0.5.0
  [739be429] MbedTLS v0.6.4
  [1d0525e4] Mill v0.1.0 [`../../Pkg/Mill.jl`]
  [20febd7b] Retry v0.3.0
  [2913bbd2] StatsBase v0.25.0
  [3bb67fe8] TranscodingStreams v0.8.1
  [228000da] XMLDict v0.2.0

**Comment by samoconnor** [comment]
> WARNING: No unread! method for HTTP.ConnectionPool.Transaction

You say that you are using the branch `so/lazyintegrate` (please note that this branch is a WIP and not intended for real use. However, I'm very pleased if you are brave enough to try it out an report issues!)

I have just done a local checkout of that branch and run `runtests.jl` without seeing that warning.

That method is defined in that branch here:
https://github.com/JuliaWeb/HTTP.jl/blob/0e3b053a52b1792a81a06e0d3d25025213405eb5/src/ConnectionPool.jl#L225-L238

Please post example code that produces the warning. My guess is that `unread!` is being called from somewhere with 2nd argument that is not a `ByteView`.

Also note that `unread!` was just removed in the `so/streamio` branch. Could you be on the wrong branch by mistake?

> Then, I sometimes occasionally get this error.
IOError(EOFError() during request(https://cta-features-jld.s3.eu-west-1.amazonaws.com/url20171004/0-part-00003.gz))
Can I prevent this?

See comments here about setting `reuse_limit=` https://github.com/JuliaWeb/HTTP.jl/issues/307#issuecomment-430938608, this may help.
If not, please post:
 - example code
 - a stack trace
 - a log with `DEBUG_LEVEL=1` in `src/HTTP.jl`

**Comment by pevnak** [comment]
Hi Sam,

downgrading everything to master seems to work, but I have not executed the full test yet, as I am fixing many bugs in my code. My desired application is machine learning, where I will have several workers downloading data from S3 and preparing minibatches, which are then sent to the master process, who use it for training. I have opted for this, because the pre-processing the data is not entirely trivial and I do not want to preprocess the data before due to the flexibility.

I was using master and so/lazyintegrate because of #307, but at the moment I will see, how it will work.

I do not understand, how should I set `reuse_limit` other than editing the source code.

I will try to post sample code next time.

Best wishes,
Tomas

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐
